File: LinkedList/Python/Deletion/deleteEND.py
class node:
    def __init__(self, data):
        self.data=data
        self.next=None

class LinkedList:
    def __init__(self):
        self.head=None

    #Insert at end of the linked list.

    def insertend(self, data):
            newnode=node(data)
            ptr=self.head
            if ptr is None:
                self.head=newnode
            else:
                while ptr.next is not None:
                    ptr=ptr.next
                ptr.next=newnode
    
    #Display the linked list.
    def display(self):
        ptr=self.head
        while ptr is not None:
            print(ptr.data, end=" -> ")
            ptr=ptr.next
        print("NULL")

    #Delete at end of the linked list.
    def deleteend(self):
        if self.head is None:
            print("The list is empty.")
        elif self.head.next is None:
            self.head = None
        else:
            ptr = self.head
            while ptr.next.next is not None:
                ptr = ptr.next
            ptr.next = None
    
head=LinkedList()
head.insertend(1)
head.insertend(2)
head.insertend(3)
head.insertend(4)
head.insertend(5)
head.display()
n=int(input("Enter how many nodes to delete: "))
for i in range(n):
    head.deleteend()
    i+=1
head.display()

    

File: LinkedList/C/Deletion/deleteEND.c
#include<stdio.h>
#include<stdlib.h>

struct node {
    int data;
    struct node *next;
};

struct node* addend(struct node *head, int data) {
    struct node *ptr;
    struct node *new_node = (struct node*)malloc(sizeof(struct node));
    new_node->data = data;
    new_node->next = NULL;
    ptr = head;
    while(ptr -> next != NULL) {
        ptr = ptr -> next;
    }
    ptr -> next = new_node;
    return head;
}

void printnode(struct node *head){
    struct node *ptr = head;
    while(ptr != NULL) {
        printf("%d -> ", ptr->data);
        ptr = ptr->next;
    }
    printf("NULL\n");
}


struct node* deleteend(struct node *head){
    if(head == NULL) {
        printf("The list is empty.\n");
    }
    else if(head->next == NULL) {
        free(head);
        head = NULL;
    }
    else {
        struct node *ptr = head;
        while(ptr->next->next != NULL) {
            ptr = ptr->next;
        }
        free(ptr->next);
        ptr->next = NULL;
    }
    return head;
}
int main() {
    int n,i;
    struct node *head = (struct node*)malloc(sizeof(struct node));
    head->data = 1;
    head->next = NULL;

    head = addend(head, 2);
    head = addend(head, 3);
    head = addend(head, 4);
    head = addend(head, 5);
    printnode(head);

    printf("Enter how many nodes to delete: ");
    scanf("%d",&n);
    for(i=0;i<n;i++){
        head= deleteend(head);
    }
    printnode(head);
    return 0;
}